File: app/src/main/java/com/kollectivemobile/euki/manager/CycleManager.java
package com.kollectivemobile.euki.manager;

import com.kollectivemobile.euki.model.CycleDayItem;
import com.kollectivemobile.euki.model.CyclePeriodData;
import com.kollectivemobile.euki.model.CyclePeriodItem;
import com.kollectivemobile.euki.networking.EukiCallback;

import java.util.List;

public interface CycleManager {
    void requestCycleItems(EukiCallback<List<CycleDayItem>> callback);
    void requestCyclePeriodData(EukiCallback<CyclePeriodData> callback);
    void deletePeriod(CyclePeriodItem item, EukiCallback<Boolean> callback);
}

File: app/src/main/java/com/kollectivemobile/euki/manager/CycleManager.kt
package com.kollectivemobile.euki.manager

import com.kollectivemobile.euki.model.CycleDayItem
import com.kollectivemobile.euki.model.CyclePeriodData
import com.kollectivemobile.euki.model.CyclePeriodItem
import com.kollectivemobile.euki.networking.EukiCallback

interface CycleManager {
  fun requestCycleItems(callback: EukiCallback<List<CycleDayItem>>)
  fun requestCyclePeriodData(callback: EukiCallback<CyclePeriodData>)
  fun deletePeriod(item: CyclePeriodItem, callback: EukiCallback<Boolean>)
}
